sdk/core/azure-core/azure/core/configuration.py

```diff
@@ -112,9 +116,18 @@ class ConnectionConfiguration(object):
             :caption: Configuring transport connection settings.
     """
 
-    def __init__(self, **kwargs):
-        self.timeout = kwargs.pop("connection_timeout", 300)
-        self.read_timeout = kwargs.pop("read_timeout", 300)
-        self.verify = kwargs.pop("connection_verify", True)
-        self.cert = kwargs.pop("connection_cert", None)
-        self.data_block_size = kwargs.pop("connection_data_block_size", 4096)
+    def __init__(
+        self,  # pylint: disable=unused-argument
+        *,
+        connection_timeout: IntOrFloat = 300,
```

Comment:
personally, prefer the explicitness of it but I have seen both used - `ThreadPoolExecutor` says int or float for timeouts while the socket library is simply `float` and the docstrings have a float value. The accompanying docstring needs to be updated since its very unclear whether its an int or a float.
`:keyword int connection_timeout: A single float in seconds for the connection timeout. Defaults to 300 seconds.`